### PR TITLE
GB-69: added new page, new test, and some driveby cleanups

### PR DESCRIPTION
**Pull request by ksadoff**
## Jira Ticket Number
https://gaming-backlog.atlassian.net/browse/GB-69

## Short description
This PR adds a new page for registering new users, plus the routing that will display the page and redirect a user to the /libraries page on creation.

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Documentation
- [ ] Other (please specify)

https://github.com/ksadoff/gaming-backlog-frontend/assets/24842433/c476c9ed-61e3-4a60-bf46-a6bdf30307b8

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,30 +1,13 @@
...
-import Game from './interfaces/Game'
 import LibrariesPage from './pages/LibrariesPage';
 import LibraryPage from './pages/LibraryPage';
 import GamePage from './pages/GamePage';
 import GameInstancePage from "./pages/GameInstancePage";
 import UserProfilePage from './pages/UserProfilePage';
+import RegistrationPage from './pages/RegistrationPage';
 
 const App = () => {
-  let emptyGame: Game = { id: "", name: "", platforms: [], genres: [], franchises: [], companies: [], releaseDate: [], summary: "", images: []}
```
Comment (ksadoff) [inline src/App.tsx:11]: This keeps causing the JSON parsing issue - we're not using it anyway, so removing

File: src/api/gameApi.ts [hunk 2]
```diff
@@ -6,14 +6,14 @@ import GameInstance from "../interfaces/GameInstance";
 
 export const getGame = async (id: string): Promise<Game> => {
     const response = await fetch(homeUrl + gamesBaseUrl + `${id}`, {
-        method: "GET",
```
Comment (ksadoff) [inline src/api/gameApi.ts:9]: Did a bunch of drivebys to remove trailing commas while debugging JSON parsing issue, seemed like a good thing to leave in even though it wasn't the cause

File: src/interfaces/Game.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
     id: string;
-    name: string 
-    platforms: Array<string>
-    genres: Array<string>
-    franchises?: Array<string>
-    companies: Array<string>
-    releaseDate: Array<string>
-    summary: string,
-    images: Array<string>
+    name: string;
```
Comment (ksadoff) [inline src/interfaces/Game.ts:3]: Drivebys to standardize our interfaces to use semicolons (preferred by TS)

File: src/App.tsx (hunk 1 above)
Comment (trauman413) [reply src/App.tsx:11]: amazing!

File: src/interfaces/Game.ts (hunk 3 above)
Comment (trauman413) [reply src/interfaces/Game.ts:3]: who needs a linter when we have you 😄 (but jk we will do that)

File: src/interfaces/UserRequest.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export default interface UserRequest {
+    displayName:string;
```
Comment (trauman413) [inline src/interfaces/UserRequest.ts]: small nit: should space this

File: src/test/unit/pages/RegistrationPage.test.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
+import { act, render, fireEvent, screen } from '@testing-library/react';
+import RegistrationPage from '../../../pages/RegistrationPage';
+import * as userApi from "../../../api/userApi";
+import { BrowserRouter } from 'react-router-dom';
+
+const registerUserStub = jest.spyOn(userApi, 'registerUser');
+const displayName = 'DoctorWho';
+const email = 'doctorwho@tardis.com';
+const password = 'twelveisthebest!';
```
Comment (trauman413) [inline src/test/unit/pages/RegistrationPage.test.tsx:9]: but ten though

**Review by trauman413 — COMMENTED** [review]
one really tiny (real) comment, looks great overall!

File: src/test/unit/pages/RegistrationPage.test.tsx (hunk 5 above)
Comment (ksadoff) [reply src/test/unit/pages/RegistrationPage.test.tsx:9]: Agree to disagree

**Review by trauman413 — APPROVED** [review]
